writer.go

```diff
@@ -16,12 +19,16 @@ type Writer interface {
 	Fprint(w io.Writer, event Event) error
 }
 
-func NewWriter(out io.Writer) Writer {
-	return &writer{out}
+func NewWriter(out io.Writer, jsonPP bool) Writer {
```

Comment:
I think if this is to make it in, the new functionality would have to be in a distinct (fully-tested) writer.

Reply:
Makes complete sense, but I don't have time to do this right now. 

I'm happy to leave this as a PR for a while, I might find time eventually 😄. Alternatively, we can close this PR and I can make a new one when I make the new writer.